### PR TITLE
chore: add typing support to regex-related adjudicators

### PR DESCRIPTION
**Pull request by samayer12**
## Description

This PR adds typing support for regex-related adjudicators. Merge this in after #1402. 

End to End Test:    
(See [Pepr Excellent Examples](https://github.com/defenseunicorns/pepr-excellent-examples))

## Related Issue

Relates to #1397, #1402

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging
- [x] Unit, [Journey](https://github.com/defenseunicorns/pepr/tree/main/journey), [E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples), [docs](https://github.com/defenseunicorns/pepr/tree/main/docs), [adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or updated as needed
- [x] [Contributor Guide Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request) followed

### CODE REVIEW

File: src/lib/capability.ts [hunk 1]
```diff
@@ -345,9 +345,9 @@ export class Capability implements CapabilityExport {
       return { ...commonChain, WithName, WithNameRegex };
     }
 
-    function InNamespaceRegex(...namespaces: RegExp[]): BindingWithName<T> {
+    function InNamespaceRegex(...namespaces: string[]): BindingWithName<T> {
```
Comment (cmwylie19) [inline src/lib/capability.ts:358]:
```suggestion
    function InNamespaceRegex(...namespaces: string[]): BindingWithName<T> {
```

This is a breaking change to the Pepr API that is in production and being used in delivery. I could be mistaken but i think we chatted about changing the regexes and decided against due to the need to serialize the bindings during the forking process

**Review by cmwylie19 — CHANGES_REQUESTED** [review]
This is a breaking change to the Pepr API that is in production and being used in delivery.

File: src/lib/capability.ts (hunk 1 above)
Comment (samayer12) [reply src/lib/capability.ts:358]: That's right, we're serializing data from a string so we can't force the data to be a `RegExp` at the moment.

**Comment by samayer12** [comment]
It has been awhile since this was created, we'll restart off of main since the journey tests are failing.

**Pull request closed without merging**